app/controllers/planning_applications/assessment/requirements_controller.rb

```diff
@@ -94,9 +94,13 @@ def set_new_requirements
         @new_requirements = current_local_authority.requirements.find(new_requirement_ids)
 
         if @new_requirements.empty?
-          redirect_to planning_application_assessment_requirements_path(@planning_application), alert: t(".missing")
+          redirect_to redirect_path, alert: t(".missing")
         end
       end
+
+      def redirect_path
+        params[:redirect_to].presence || planning_application_assessment_requirements_path(@planning_application)
```

Comment:
This sort of logic might end up being needed in a few places, but is it possible to do it without `params`? Maybe `session` is safer, but I don't know if that works. (That's what the flash messages use, as I understand?)

Reply:
Example in #2679. (room for improvement perhaps: open to suggestions.)